style/java.md

```diff
@@ -0,0 +1,33 @@
+# Java
+
+At Khan Academy, we only use Java for Android and, as such, follow the [Android Code Style Guidelines for Contributors](https://source.android.com/source/code-style.html). Note that those rules themselves extend the [standard Java language style guide as of 1999](http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141855.html). The rest of this document describes additions and clarifications to those rules that we follow at Khan Academy.
```

Comment:
> we only use Java for Android

One minor exception that I can think of:
https://github.com/Khan/ka-pycharm-plugin/blob/master/src/org/khanacademy/pycharmplugin/KATestConfigurationProducer.java

Not a whole lot there, but that's what my Java looks like normally.